lib/ethpy/ethpy/test_fixtures/deploy_hyperdrive.py

```diff
@@ -0,0 +1,242 @@
+"""Functions to initialize hyperdrive using web3"""
+
+from eth_account.account import Account
+from eth_account.signers.local import LocalAccount
+from ethpy.base import (
+    deploy_contract,
+    deploy_contract_and_return,
+    get_transaction_logs,
+    initialize_web3_with_http_provider,
+    load_all_abis,
+    smart_contract_transact,
+)
+from fixedpointmath import FixedPoint
+from web3 import Web3
+from web3.contract.contract import Contract
+
+# TODO these functions should eventually be moved to `ethpy/hyperdrive`, but leaving
+# these here for now to be used by tests while we figure out how to parameterize
+# initial hyperdrive conditions
+
+
+# Following solidity implementation here, so matching function name
+def _calculateTimeStretch(apr: int) -> int:  # pylint: disable=invalid-name
+    """Helper function mirroring solidity calculateTimeStretch
+
+    Arguments
+    --------
+    apr: int
+        The scaled input apr
+
+    Returns
+    -------
+    int
+        The scaled output time stretch
+    """
+    fp_apr = FixedPoint(scaled_value=apr)
+    time_stretch = FixedPoint(scaled_value=int(5.24592e18)) / (
+        FixedPoint(scaled_value=int(0.04665e18)) * (fp_apr * 100)
+    )
+    return (FixedPoint(scaled_value=int(1e18)) / time_stretch).scaled_value
+
+
+def initialize_deploy_account(web3: Web3) -> LocalAccount:
+    """Initializes the local anvil account to deploy everything from.
+
+    Arguments
+    --------
+    web3 : Web3
+        web3 provider object
+
+    Returns
+    -------
+    LocalAccount
+        The LocalAccount object
+    """
+    # TODO get private key of this account programatically
```

Comment:
I don't think this is possible :( I was trying to do it a while back. 

However, I think one solution is to force the [mnemonic](https://book.getfoundry.sh/reference/anvil/#:~:text=%2Dm%2C%20%2D%2Dmnemonic%20%3CMNEMONIC%3E) when you spin up anvil, and then you can use web3 to generate a private key from that mnemonic (`account = w3.eth.account.from_mnemonic("some dumb list of words"`). This should be the same private key as the one anvil is using. The mnemonic can be set in a `.env` file.

Reply:
You can leave this as a TODO. I made an [issue for it](https://github.com/delvtech/elf-simulations/issues/816)